File: Java/Tema7_Ejercicio7.java
/*
 * TEMA 7: FUNCIONES Y MÉTODOS - EJERCICIO 7
 * ==========================================
 * Escriba un programa que verifique si una cadena es un palíndromo usando función.
 */

public class Tema7_Ejercicio7 {
    
    static boolean esPalindromo(String texto) {
        String limpio = texto.replaceAll("\\s+", "").toLowerCase();
        String invertido = new StringBuilder(limpio).reverse().toString();
        return limpio.equals(invertido);
    }
    
    static int contarVocales(String texto) {
        int contador = 0;
        for (char c : texto.toLowerCase().toCharArray()) {
            if ("aeiou".indexOf(c) != -1) {
                contador++;
            }
        }
        return contador;
    }
    
    public static void main(String[] args) {
        System.out.println("=== FUNCIONES CON CADENAS ===");
        
        String[] textos = {"Anita lava la tina", "Hola mundo", "Reconocer", "Java"};
        
        for (String texto : textos) {
            System.out.println("\nTexto: '" + texto + "'");
            System.out.println("  ¿Es palíndromo? " + esPalindromo(texto));
            System.out.println("  Vocales: " + contarVocales(texto));
        }
    }
}

/*
 * SALIDA ESPERADA:
 * === FUNCIONES CON CADENAS ===
 * 
 * Texto: 'Anita lava la tina'
 *   ¿Es palíndromo? true
 *   Vocales: 6
 * 
 * Texto: 'Hola mundo'
 *   ¿Es palíndromo? false
 *   Vocales: 4
 * 
 * Texto: 'Reconocer'
 *   ¿Es palíndromo? true
 *   Vocales: 3
 * 
 * Texto: 'Java'
 *   ¿Es palíndromo? false
 *   Vocales: 2
 */

File: JS/Tema7_Ejercicio7.js
/*
 * TEMA 7: FUNCIONES Y MÉTODOS - EJERCICIO 7
 * =========================================
 * Funciones de manipulación de strings.
 * 
 * EXPLICACIÓN:
 * - Contar vocales y consonantes
 * - Verificar palíndromo
 * - Truncar texto
 */

function contarVocales(texto) {
    const vocales = texto.toLowerCase().match(/[aeiou]/g);
    return vocales ? vocales.length : 0;
}

function esPalindromo(texto) {
    const limpio = texto.toLowerCase().replace(/[^a-z0-9]/g, '');
    return limpio === limpio.split('').reverse().join('');
}

function truncarTexto(texto, longitud = 20, sufijo = '...') {
    if (texto.length <= longitud) return texto;
    return texto.substring(0, longitud) + sufijo;
}

function invertirPalabras(texto) {
    return texto.split(' ').reverse().join(' ');
}

console.log("=== MANIPULACIÓN DE STRINGS ===");

const texto = "JavaScript es increíble";
console.log(`Texto: "${texto}"`);
console.log(`Vocales: ${contarVocales(texto)}`);

console.log(`\nPalíndromos:`);
const palabras = ['radar', 'mundo', 'level', 'hello'];
palabras.forEach(p => {
    console.log(`"${p}": ${esPalindromo(p)}`);
});

console.log(`\nTruncamiento:`);
const largo = "Este es un texto muy largo que necesita ser truncado";
console.log(`Original: "${largo}"`);
console.log(`Truncado: "${truncarTexto(largo, 30)}"`);

console.log(`\nInversión de palabras:`);
console.log(`Original: "${texto}"`);
console.log(`Invertido: "${invertirPalabras(texto)}"`);

/*
 * SALIDA ESPERADA:
 * === MANIPULACIÓN DE STRINGS ===
 * Texto: "JavaScript es increíble"
 * Vocales: 8
 * 
 * Palíndromos:
 * "radar": true
 * "mundo": false
 * "level": true
 * "hello": false
 * 
 * Truncamiento:
 * Original: "Este es un texto muy largo que necesita ser truncado"
 * Truncado: "Este es un texto muy largo ..."
 * 
 * Inversión de palabras:
 * Original: "JavaScript es increíble"
 * Invertido: "increíble es JavaScript"
 */
